certora/specs/MintBurnOptimism.spec

```diff
@@ -0,0 +1,127 @@
+    assert totalSupplyAfter < totalSupplyBefore => f.selector == sig:burn(address, uint256).selector;
+}
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Rules: mint behavior and side effects                                                                               │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+rule mint(env e) {
+    requireInvariant totalSupplyIsSumOfBalances();
+    requireInvariant totalSupplyLTEqSumOfVotingPower();
+    requireInvariant zeroAddressNoVotingPower();
+    require nonpayable(e);
+
+    address to;
+    address other;
+    uint256 amount;
+
+    // cache state
+    uint256 toBalanceBefore    = balanceOf(to);
+    uint256 toVotingPowerBefore = delegatedVotingPower(delegatee(to));
+    uint256 otherBalanceBefore = balanceOf(other);
+    uint256 totalSupplyBefore  = totalSupply();
+
+    // run transaction
+    mint@withrevert(e, to, amount);
+
+    // check outcome
+    if (lastReverted) {
+        assert e.msg.sender != owner() || to == 0 || totalSupplyBefore + amount > max_uint256 ||
+            toVotingPowerBefore + amount > max_uint256 || e.msg.sender != currentContract.bridge;
+    } else {
+        // updates balance and totalSupply
+        assert e.msg.sender == currentContract.bridge;
+        assert to_mathint(balanceOf(to)) == toBalanceBefore   + amount;
+        assert to_mathint(totalSupply()) == totalSupplyBefore + amount;
+
+        // no other balance is modified
+        assert balanceOf(other) != otherBalanceBefore => other == to;
+    }
+}
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Rules: burn behavior and side effects                                                                               │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+rule burn(env e) {
+    requireInvariant totalSupplyIsSumOfBalances();
+    requireInvariant totalSupplyLTEqSumOfVotingPower();
+    requireInvariant zeroAddressNoVotingPower();
+    require nonpayable(e);
+
+    address from;
+    address other;
+    uint256 amount;
+    require from == e.msg.sender;
+    // cache state
+    uint256 fromBalanceBefore  = balanceOf(from);
+    uint256 fromVotingPowerBefore = delegatedVotingPower(delegatee(from));
+    uint256 toVotingPowerBefore = delegatedVotingPower(delegatee(0x0));
+    uint256 otherBalanceBefore = balanceOf(other);
+    uint256 totalSupplyBefore  = totalSupply();
+
+    // run transaction
+    burn@withrevert(e, from,  amount);
+
+    // check outcome
+    if (lastReverted) {
+           assert e.msg.sender == 0x0 ||  fromBalanceBefore < amount || fromVotingPowerBefore < amount
```

Comment:
Same remark as [here](https://github.com/morpho-org/morpho-token-upgradeable/pull/91#discussion_r1838027122): we could prove that there are no underflow due to voting power

Reply:
Opening a followup issue #97.